### PR TITLE
added categories table for products and endpoint for getting the menu

### PR DESCRIPTION
**Pull request by VeniNovakov**
More of a rework than the whole feature itself

### CODE REVIEW

File: pizzeria-backend/Program.cs [hunk 1]
```diff
@@ -18,7 +18,8 @@
 
 // Add services to the container.
 
-builder.Services.AddControllers();
+builder.Services.AddControllers().AddNewtonsoftJson(options =>
+    options.SerializerSettings.ReferenceLoopHandling = Newtonsoft.Json.ReferenceLoopHandling.Ignore);
```
Comment (GenchoBG) [inline pizzeria-backend/Program.cs:22]: I saw this on the other MR and I was wondering - why do you need this?

File: pizzeria-backend/Controllers/ProductsController.cs [hunk 2]
```diff
@@ -68,6 +41,18 @@ public async Task<IActionResult> GetProduct(int id)
...
         }
 
+        [HttpGet("menu")]
+        [Produces("application/json")]
+        public async Task<IActionResult> GetMenu()
+        {
+            var menu = await _productService.GetMenu();
+            if (menu == null)
+            {
+                return NotFound("message: " + "There are no items in the menu");
```
Comment (GenchoBG) [inline pizzeria-backend/Controllers/ProductsController.cs]: Why is this "message: " needed?

File: pizzeria-backend/Program.cs (hunk 1 above)
Comment (VeniNovakov) [reply pizzeria-backend/Program.cs:22]: Infinite reference loops were happening when using .Include to load the needed relationships

File: pizzeria-backend/Controllers/ProductsController.cs (hunk 2 above)
Comment (VeniNovakov) [reply pizzeria-backend/Controllers/ProductsController.cs]: Not needed, only had a thought it would look good

Comment (VeniNovakov) [reply pizzeria-backend/Controllers/ProductsController.cs]: Done

**Pull request merged**